pkg/api/roles.go

```diff
@@ -194,29 +196,48 @@ func (hs *HTTPServer) declareFixedRoles() error {
 		Grants: []string{string(accesscontrol.RoleGrafanaAdmin)},
 	}
 
-	teamWriterGrants := []string{string(models.ROLE_ADMIN)}
-	if hs.Cfg.EditorsCanAdmin {
-		teamWriterGrants = append(teamWriterGrants, string(models.ROLE_EDITOR))
-	}
 	teamsWriterRole := accesscontrol.RoleRegistration{
 		Role: accesscontrol.RoleDTO{
 			Name:        "fixed:teams:writer",
 			DisplayName: "Team writer",
+			Description: "Read and manage team memberships.",
+			Group:       "Teams",
+			Version:     2,
+			Permissions: []accesscontrol.Permission{
+				{
+					Action: ActionTeamsPermissionsRead,
+				},
+				{
+					Action: ActionTeamsPermissionsWrite,
+				},
+			},
+		},
+		Grants: []string{string(models.ROLE_ADMIN)},
+	}
+
+	teamCreatorGrants := []string{string(models.ROLE_ADMIN)}
+	if hs.Cfg.EditorsCanAdmin {
+		teamCreatorGrants = append(teamCreatorGrants, string(models.ROLE_EDITOR))
+	}
+	teamsCreatorRole := accesscontrol.RoleRegistration{
```

Comment:
A separate role for team creation is needed, as `editorsCanAdmin` option allows editors to create new teams, but doesn't allow them to manage the existing teams.